urllib3/util/retry.py

```diff
@@ -295,6 +294,20 @@ def __repr__(self):
                 'read={self.read}, redirect={self.redirect})').format(
                     cls=type(self), self=self)
 
+    def redirect_method(self, method, status):
+        """
+        Assuming we're doing a redirect, should we change HTTP methods?
+        """
+        if method == 'GET':
+            return method
+        if method == 'HEAD':
+            return method
+        if status == 303:
+            return 'GET'
+        if method == 'POST' and status in set([301, 302, 303]):
```

Comment:
I'm hoisting this to module scope, but leaving 303 in the set to help future readers. Explicit (having the code in the set) is better than implicit (skip it, because the logic flow prevents it from being a factor).